poh/src/poh_recorder.rs

```diff
@@ -432,7 +432,7 @@ impl PohRecorder {
         self.leader_bank_notifier.clone()
     }
 
-    fn is_same_fork_as_previous_leader(&self, slot: Slot) -> bool {
+    fn start_slot_was_mine_or_previous_leader(&self, slot: Slot) -> bool {
```

Comment:
I'm thinking this should just be "start slot was previous leader" even though it would technically return true if we happen to be starting from our own slot.

Rationale is:
- If it's actually our block, we would want to skip grace ticks. 
- Immediately before this, we check the start bank collector ID to determine if it was ours, so practically speaking, it shouldn't return true due to matching on our slot.

Reply:
I'm more in favor of explicitly documenting the true behavior to avoid misuse of this method in the future. It's true that the current usage effectively means "start slot was previous leader" but that depends on first calling `fn start_slot_was_mine`. If `fn start_slot_was_mine` isn't called first, it's possible that we return false for `fn can_skip_grace_ticks` when building off our own slot because `fn start_slot_was_mine_or_previous_leader` would be true and `fn building_off_previous_leader_last_block` would be false.

Reply:
Yeah I'll update that param name, it was bothering me too

Reply:
Updated to `next_leader_slot` everywhere